assets/js/components/wp-dashboard/WPDashboardPopularPages.js

```diff
@@ -109,16 +111,16 @@ export default function WPDashboardPopularPages( props ) {
 		return <WidgetReportError moduleSlug="analytics" error={ error } />;
 	}
 
-	if ( isGatheringData && ! zeroDataStatesEnabled ) {
+	if (
+		! zeroDataStatesEnabled &&
+		isGatheringData &&
+		isZeroReport( report )
+	) {
 		return <WidgetReportZero moduleSlug="analytics" />;
 	}
 
-	// Skip rendering the table if there are no rows.
-	if ( ! report[ 0 ].data?.rows?.length ) {
-		return null;
-	}
-
-	const rows = cloneDeep( report[ 0 ].data.rows );
+	// data.rows is not guaranteed to be set so we need a fallback.
+	const rows = cloneDeep( report[ 0 ].data.rows ) || [];
```

Comment:
Won't this still cause a JS error if e.g. `data` or `rows` is `undefined`?

Reply:
@aaemnnosttv Just tested and see there's no error. However, I feel like it's unnecessary to even `cloneDeep` then, maybe we should wrap it accordingly if `data` or `data.rows` is `undefined`, we can immediately fall back to `[]` then.

Reply:
From what I recall, `data` is always defined, it's the `rows` property which may be `undefined`.

Here's the actual `report` in the console when gathering data:
<img width="420" alt="image" src="https://user-images.githubusercontent.com/1621608/160479726-7f3a42f4-309c-45db-87d5-ba6cd4b1df29.png">

Reply:
I just tested with and without the ZDS flag and empty cache as well. `data` is always set because we have guards above for `loading` and `error` cases above.

Reply:
Sorry I missed your second comment – 

> However, I feel like it's unnecessary to even `cloneDeep` then, maybe we should wrap it accordingly if `data` or `data.rows` is `undefined`, we can immediately fall back to `[]` then.

I took a look in `cloneDeep` and it's not much overhead to call it with `undefined` (it returns the given value if it's not an object), but we can make that call conditional easily enough, I'll make that change.